azure/services/inboundnatrules/inboundnatrules.go

```diff
@@ -56,9 +56,14 @@ func New(scope InboundNatScope) *Service {
 
 // Reconcile gets/creates/updates an inbound NAT rule.
 func (s *Service) Reconcile(ctx context.Context) error {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "inboundnatrules.Service.Reconcile")
+	ctx, log, done := tele.StartSpanWithLogger(ctx, "inboundnatrules.Service.Reconcile")
 	defer done()
 
+	if s.Scope.APIServerLBName() == "" {
```

Comment:
How is it possible to create a cluster without a apiserver load balancer? In other words, how does the above condition (empty string value for `AzureCluster.Spec.NetworkSpec.APIServerLB.Name`) ever occur?

Reply:
@jackfrancis as mentioned in the PR description, this can happen if the cluster is managed externally, see https://capz.sigs.k8s.io/topics/externally-managed-azure-infrastructure.html

Reply:
Thanks. The code lgtm. I am a bit confused about this line in the docs:

> If the AzureCluster resource includes a “cluster.x-k8s.io/managed-by” annotation then the [controller will skip any reconciliation](https://cluster-api.sigs.k8s.io/developer/providers/cluster-infrastructure.html#normal-resource).

If reconciliation is skipped, then why is this change needed?

Reply:
The cluster controllers skips reconciling it, however the azuremachine-related controllers still do: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/a42074cdc1e82bb093ec2c79d99924bf5ef726dc/controllers/azuremachine_reconciler.go#L90-L92

We want to use capz to only manage worker machines, which is how we ran into this.

Reply:
cc @Jont828

Reply:
@Jont828 we were talking offline yesterday about your wanting this scenario to `s.Scope.UpdatePutStatus` even if it's a no-op (no LB present). Can you explain why?

Reply:
Agree that setting the condition when there is no resource is confusing which is why we have #1868 open to address that. However in the meantime we need to set all conditions that are defined in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/scope/machine.go#L550 to true otherwise the AzureMachine won't reach Ready condition as summary will include false conditions. However in this case it looks like `InboundNATRulesReadyCondition` is missing from the list of summary conditions (potentially a bug?).

Reply:
I see, updated the change to include `UpdatePutStatus` even when there is no LB